astro/src/content/blog/building-self-hostable-application.mdx

```diff
@@ -0,0 +1,366 @@
+
+Unless you go with the Chrome auto-upgrade model, you'll have customers and users using different versions. Think about how to support them. How old a version will you support? What if a customer is paying you a lot of money?
+
+#### LTS
+
+With a self-hostable product, you should create a long term support (LTS) program like Linux distributions or programming languages. An LTS lets developers using your tool upgrade to certain versions and remain on them for a long period of time. Critical self-hostable products share a lot of similarities with programming languages. Your users may be hesitant to invest to regularly upgrade your product. Features can encourge it, but once your product meets their needs, they may resist upgrading.
+
+Since everyone is not on the same version, backporting bug fixes becomes a discussion. This increases development complexity. This includes security fixes too. What will you backport only to LTSes? 
+
+## Customer Enablement
+
+The above are all considerations you and your engineering team must consider when building a self-hostable product. But other than these technical considerations, what else does your customer need to succeed?
+
+Operating software can be difficult. That is why many people prefer SaaS solutions. But teams self-host other parts of their application, whether in a cloud or on their own hardware. These teams operate software at the scale and performance their employer needs. If you are selling a self-hostable product, enable your customer with great documentation.
+
+This includes detailed documentation around:
+
+* installation
+* scaling
+* performance tuning
+* monitoring
+* troubleshooting
+* system requirements
+* infrastructure integrations
+* tooling integrations
+* upgrades
+
+You'll need to make this as thorough, accurate and complete as you can. You'll also need to clearly tag them with versions of your software, because capabilities and requirements change as your software evolves.
+
+The number of possible combinations of how your software can be installed or operated is nearly infinite. Try to limit the effort by documenting common integrations and complementary software components.
+
+Clearly document your changes too, as part of your release notes. You can do this in lo-fi ways such as semver or high fidelity ways like old-style release notes. 
+
+### Configuration Management
+
+The type of organization that can confidently self-host critical infrastructure also leverage configuration management. They want to configure your application via an API or tooling like Terraform or Pulumi. 
+
+Offering this functionality allows your customers to manage changes to your product in the same way they manage other parts of their application infrastructure.
```

Comment:
Seems like another place you may mention kickstarts.

Reply:
Kickstarts don't let you do this, though.

What am I missing?

Reply:
I was just thinking about spinning up FusionAuth into a certain state, re-reading the context it may not make sense here.